### PR TITLE
migrated user spec to playwright

### PR DESCRIPTION
**Pull request by Ashish8689**
### Describe your changes:

- I worked on migrating user spec to playwright

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/Users.spec.ts [hunk 1]
```diff
@@ -10,46 +10,503 @@
...
+        await settingClick(dataConsumerPage, id.testid);
+      } else {
+        await sidebarClick(dataConsumerPage, SidebarItem.SETTINGS);
+        let paths = SETTINGS_OPTIONS_PATH[id.testid];
+
+        if (id.isCustomProperty) {
+          paths = SETTING_CUSTOM_PROPERTIES_PATH[id.testid];
+        }
+
+        // eslint-disable-next-line jest/no-conditional-expect
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/Users.spec.ts:328]: why we need this?

File: openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/Users.spec.ts [hunk 2]
```diff
@@ -10,46 +10,503 @@
...
+      } else {
+        await sidebarClick(dataStewardPage, SidebarItem.SETTINGS);
+        let paths = SETTINGS_OPTIONS_PATH[id.testid];
+
+        if (id.isCustomProperty) {
+          paths = SETTING_CUSTOM_PROPERTIES_PATH[id.testid];
+        }
+
+        // eslint-disable-next-line jest/no-conditional-expect
+        await expect(dataStewardPage.getByTestId(paths[0])).not.toBeVisible();
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/Users.spec.ts:464]: same?

File: openmetadata-ui/src/main/resources/ui/playwright/utils/user.ts [hunk 3]
```diff
@@ -168,3 +186,520 @@ export const hardDeleteUserProfilePage = async (
...
+
+  const updateDescription = page.waitForResponse('/api/v1/users/*');
+  await page.click('[data-testid="save"]');
+  await updateDescription;
+
+  await page.click('.ant-collapse-expand-icon > .anticon > svg');
+
+  // Verify the updated description
+  const description = page.locator(
+    ':nth-child(2) > :nth-child(1) > [data-testid="viewer-container"] > [data-testid="markdown-parser"] > :nth-child(1) > .toastui-editor-contents > p'
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/utils/user.ts:240]: i think, we need to re-think here for selectors! we have description const can you check that?

File: openmetadata-ui/src/main/resources/ui/playwright/utils/user.ts [hunk 4]
```diff
@@ -168,3 +186,520 @@ export const hardDeleteUserProfilePage = async (
...
+  // Check edit description permission
+  await expect(page.locator('[data-testid="edit-description"]')).toBeVisible();
+
+  // Check edit tier permission
+  await expect(page.locator('[data-testid="edit-tier"]')).toBeVisible();
+
+  // Check add tags button
+  await expect(
+    page.locator(
+      ':nth-child(2) > [data-testid="tags-container"] > [data-testid="entity-tags"] > .m-t-xss > .ant-tag'
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/utils/user.ts:516]: can you please filter the selection

File: openmetadata-ui/src/main/resources/ui/playwright/utils/user.ts [hunk 5]
```diff
@@ -168,3 +186,520 @@ export const hardDeleteUserProfilePage = async (
...
+    )
+  ).toBeVisible();
+
+  // Check add glossary term button
+  await expect(
+    page.locator(
+      ':nth-child(3) > [data-testid="glossary-container"] > [data-testid="entity-tags"] > .m-t-xss > .ant-tag'
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/utils/user.ts:523]: same

File: openmetadata-ui/src/main/resources/ui/playwright/utils/user.ts [hunk 6]
```diff
@@ -168,3 +186,520 @@ export const hardDeleteUserProfilePage = async (
...
+  await expect(page.locator('[data-testid="edit-description"]')).toBeVisible();
+
+  // Check edit tier permission
+  await expect(page.locator('[data-testid="edit-tier"]')).toBeVisible();
+
+  // Check add tags button
+  await expect(
+    page.locator(
+      ':nth-child(2) > [data-testid="tags-container"] > [data-testid="entity-tags"] > .m-t-xss > .ant-tag'
+    )
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/utils/user.ts:613]: same

File: openmetadata-ui/src/main/resources/ui/playwright/utils/user.ts [hunk 7]
```diff
@@ -168,3 +186,520 @@ export const hardDeleteUserProfilePage = async (
...
+  ).toBeVisible();
+
+  // Check add glossary term button
+  await expect(
+    page.locator(
+      ':nth-child(3) > [data-testid="glossary-container"] > [data-testid="entity-tags"] > .m-t-xss > .ant-tag'
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/utils/user.ts:619]: please check everywhere

File: openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/Users.spec.ts [hunk 8]
```diff
@@ -10,46 +10,503 @@
...
+  await tableEntity.delete(apiContext);
+  await tableEntity2.delete(apiContext);
+  await policy.delete(apiContext);
+  await role.delete(apiContext);
+
+  await afterAction();
+});
+
+test.describe('User with Admin Roles', () => {
+  test.slow(true);
```
[the remark this answers is not in the document]
Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/Users.spec.ts:131]: there were multiple tests was failing due to timeout on each run, that the  reason have used slow in the describe itself!

**Review by ShaileshParmar11 — APPROVED** [review]

**Pull request merged**